lib/article_rating_extractor.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+#= Fetches article classes from PageAssessments response
+# We simplify this parser by folding the nonstandard ratings
+# into the corresponding standard ones. We don't want to deal with edge cases
+# like bplus and a/ga.
+
+class ArticleRatingExtractor
+  def initialize(pageassessments)
+    @assessments = pageassessments || {}
+  end
+
+  ################
+  # Entry point #
+  ################
+
+  def ratings
+    # Returns the classes of articles mapped to the respective title
+    ratings = {}
+    @assessments.values.each do |page|
+    # The class of the first pageassessment is taken
+      rating =  ratings[page['title'].tr(' ', '_')]
+      ratings[page['title'].tr(' ', '_')] ||= get_article_class(page['pageassessments'])
+    end
+    return ratings
+  end
+
+  private
+
+  def get_article_class(pageassessments)
+    # Returns nil if no page assessment data found
+    return nil if pageassessments.nil?
+    return nil unless pageassessments.is_a? Enumerable
+    pageassessments.map do |project, info|
```

Comment:
Doing a 'map' here is a little confusing, since we only care about the first value, and we definitely don't care about the return value. We could instead do something like `pagesassessments.values.first` to get the wikiproject we want to extract the class from.